src/VisualizationTests/HelixRenderPackageTests.cs

```diff
@@ -157,6 +158,398 @@ public void HelixRenderPackage_MeshColorBufferCheck()
             Assert.AreEqual(p.MeshVertexColors, testColors);
         }
 
+        [Test]
+        public void HelixRenderPackage_AppendMeshVertexColorRange_Success()
+        {
+            var p = new HelixRenderPackage();
+
+            PushQuadIntoPackage(p);
+
+            var testColors = WhiteByteArrayOfLength(6);
+
+            p.AppendMeshVertexColorRange(testColors);
+
+            Assert.AreEqual(p.MeshVertexColors, testColors);
+            
+            PushQuadIntoPackage(p);
+            
+            p.AppendMeshVertexColorRange(testColors);
+
+            var totalColors = WhiteByteArrayOfLength(12);
+            
+            Assert.AreEqual(p.MeshVertexColors, totalColors);
+        }
+
+        [Test]
+        public void HelixRenderPackage_AppendMeshVertexColorRange_ThrowsForBadRange()
+        {
+            var p = new HelixRenderPackage();
+
+            PushQuadIntoPackage(p);
+
+            var testColors = WhiteByteArrayOfLength(5);
+
+            try
+            {
+                p.AppendMeshVertexColorRange(testColors);
+            }
+            catch(Exception e)
+            {
+                Assert.AreEqual(e.GetType(), typeof(Exception));
+            }
```

Comment:
`Assert.ThrowsException<Exception>(() => p.AppendMeshVertexColorRange(testColors));`

Reply:
How are these exceptions dealt with in Dynamo in a real situation?

Reply:
All the exceptions (my new ones and the old ones) within the `RenderPacakge` implementation are caught here -> https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Scheduler/UpdateRenderPackageAsyncTask.cs#L303.  The net affect is that the node does not display geometry in the background preview.

Reply:
The only difference in the new implementation is the specific handling of the `LegacyRenderPackageMethodException